packages/react-native-client/ios/VideoPreviewView.swift

```diff
@@ -26,12 +27,28 @@ class VideoPreviewView: ExpoView {
                 return
             }
 
-            localVideoTrack =
+            self.localVideoTrack =
                 tracks.first(where: { (key, track) in
                     track is LocalVideoTrack
                 })?.value as? LocalVideoTrack
-            localVideoTrack?.start()
-            videoView?.track = localVideoTrack
+            self.localVideoTrack?.start()
+            self.videoView?.track = self.localVideoTrack
+        }
+    }
+
+    override func willMove(toSuperview newSuperview: UIView?) {
+        super.willMove(toSuperview: newSuperview)
+        if newSuperview == nil {
+            RNFishjamClient.localCameraTrackListeners.removeAll(where: {
+                if let view = $0 as? VideoRendererView {
```

Comment:
**issue:** 
Shouldn't this be `VideoRendererView`? 
```suggestion
                if let view = $0 as? VideoPreviewView {
```

Also I believe you don't need to cast anything here. 
If you add `protocol OnLocalCameraTrackChangedListener: AnyObject` 

`RNFishjamClient.localCameraTrackListeners.removeAll(where: { $0 === self })` should work.

Reply:
You are right in both cases, i will update protocol declaration.